File: 1.HTTP-AND-CORS/code/typescript/the_exchange_step_by_step_12.ts
import express, { Request, Response } from 'express';
import path from 'path';

const app = express();

// express.static or res.sendFile handles Range, 206, 416 and If-Range for you,
// driven by the file's modtime and an optional ETag.
app.get('/download', (req: Request, res: Response) => {
    const filePath = path.resolve(__dirname, 'big.zip');
    res.sendFile(filePath);
});

File: 1.HTTP-AND-CORS/code/rust/the_exchange_step_by_step_12.rs
use actix_web::{HttpRequest, Responder};
use actix_files::NamedFile;
use std::path::PathBuf;

// actix_files handles Range, 206, 416 and If-Range for you,
// driven by the file's modtime and an optional ETag.
async fn download(req: HttpRequest) -> actix_web::Result<impl Responder> {
    let path: PathBuf = "big.zip".parse().unwrap();
    Ok(NamedFile::open(path)?.into_response(&req))
}
